pkg/sdk/sdk.go

```diff
@@ -0,0 +1,147 @@
+package sdk
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/PereRohit/util/log"
+	respModel "github.com/PereRohit/util/model"
+	"github.com/gorilla/mux"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"time"
+)
+
+type htmlToPdfSvc struct {
+	svcUrl string
+}
+
+func NewHtmlToPdfSvc(url string) HtmlToPdfSvcI {
+	return &htmlToPdfSvc{
+		svcUrl: url,
+	}
+}
+
+type HtmlToPdfSvcI interface {
+	Register([]byte) (string, error)
+	Replace([]byte, string) error
+	GeneratePdf(map[string]interface{}, string) ([]byte, error)
+}
+
+func (h *htmlToPdfSvc) Register(fileBytes []byte) (string, error) {
```

Comment:
All the below test scenarios need to be covered:
1. Post failure
2. non 200 status code received
3. ReadAll failure
4. Unmarshal failure
5. map inference failure
6. Success case

Reply:
I have included 5 test cases but I was unable to fail readall case

Reply:
cannot fail the readall as server doesnot allow response body as nil

Reply:
Cases 1 & 5 are yet to be covered.

Reply:
Which test cases points to which scenario for 1 & 5?

Reply:
done